tesseract/tesseract_scene_graph/include/tesseract_scene_graph/joint.h

```diff
@@ -256,6 +256,24 @@ class Joint
     this->type = JointType::UNKNOWN;
   }
 
+  /* Create a shallow copy of current joint, with prefix prepended to its name, child link name and parent link name.
+   * Shallow means that underlying properties, such as dynamics, limits... are shared with the original object.*/
+  Joint prefix(const std::string& prefix) const
+  {
```

Comment:
With this being a shallow means that when one changes they all change, is this the desired affect? Is there a down side with making this a deep copy? If this is needed there is some more code throughout that would need to be tested and updated like rviz plugins,etc. because I am not sure if you update one it would update them all.

Reply:
Yes that's the idea, every prefixed version of the Joint will share (via `shared_ptr`) the same attributes. I implemented it this way because:
- It was easier
- I did not want to copy Link geometry, and applied the same logic to Joint
But I understand your point... I can make it a deep copy easily, by copying the underlying objects instead of the pointers.

Reply:
I agree with your point. The geometry has a clone method which does an efficient copy. The geometry data are stored in shared pointer where it copies the shared pointer on clone. This is ok because there is no way to change the geometry data from the classes interface.

It would be good to add a clone method the objects in link.h and joint.h. Also would be good to delete the copy and assignment constructor.

Reply:
Please let me know what you would prefer, I'll update here.

Reply:
I think it would be good to implement the clone method and delete the copy and assignment constructor for the types in link.h and joint.h. I think the only class that does not need a clone is the material.

Reply:
Then your prefix function for link and joint would become the clone function that takes a name.